platform/commonUI/edit/src/actions/EditAndComposeAction.js

```diff
@@ -54,9 +54,13 @@ define(
                         .then(doPersist);
             }
 
+            if (this.domainObject.getCapability('type').getKey() !== 'folder') {
+                this.domainObject.getCapability('action').perform('edit');
+            }
```

Comment:
Two things:

* I think performing edit after linking is backwards; we should initiate edit mode, then link, so that the user can Cancel.
* Is there a better check than "is it a folder" for this?

First thing that comes to mind, which would solve both, is to check for the existence of an `edit` action. If one exists, perform it, and then add the link. Otherwise, just add the link.